Civi/Api4/Generic/BasicGetFieldsAction.php

```diff
@@ -144,6 +146,9 @@ protected function formatResults(&$values, $isInternal) {
       if (isset($defaults['options'])) {
         $this->formatOptionList($field);
       }
+      if ($this->getAction() === 'create' && $field['readonly'] === TRUE) {
+        $field['input_type'] = 'DisplayOnly';
+      }
       $field = array_diff_key($field, $internalProps);
     }
     // Hide the 'contact_type' field from Individual,Organization,Household pseudo-entities
```

Comment:
I still think adding `'DisplayOnly' => ts('Display Only'),` to L336 would be good.